switch_model/generators/extensions/storage.py

```diff
@@ -7,6 +7,7 @@
 storage, when to charge, energy accounting, etc.
 """
 import math
+import scipy
```

Comment:
I was curious where you are using `scipy`. It looks like you only use one function of it. Consider replacing it with

```
from scipy.fft import fftfreq
```